modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java

```diff
@@ -716,6 +728,92 @@ public String getProcesses() throws ProcessCenterException {
         }
     }
 
+    public String getSucessorPredecessorSubprocessList(String resourcePath){
+        String resourceString = "";
+        try{
+            RegistryService registryService = ProcessCenterServerHolder.getInstance().getRegistryService();
+            if (registryService != null) {
+                UserRegistry reg = registryService.getGovernanceSystemRegistry();
+                resourcePath = resourcePath.substring("/_system/governance/".length());
+                Resource resourceAsset = reg.get(resourcePath);
+                String resourceContent = new String((byte[]) resourceAsset.getContent());
+
+                JSONObject conObj = new JSONObject();
+                DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+                DocumentBuilder builder;
+                builder = factory.newDocumentBuilder();
+                Document document = builder.parse(new InputSource(new StringReader(resourceContent)));
+
+                JSONArray subprocessArray = new JSONArray();
+                JSONArray successorArray = new JSONArray();
+                JSONArray predecessorArray = new JSONArray();
+
+                conObj.put("subprocesses", subprocessArray);
+                conObj.put("successors", successorArray);
+                conObj.put("predecessors", predecessorArray);
+
+                NodeList subprocessElements = ((Element)document.getFirstChild()).getElementsByTagName("subprocess");
+                NodeList successorElements = ((Element)document.getFirstChild()).getElementsByTagName("successor");
+                NodeList predecessorElements = ((Element)document.getFirstChild()).getElementsByTagName("predecessor");
+
+                if(subprocessElements.getLength() != 0){
+                    for (int i = 0 ; i < subprocessElements.getLength() ; i++) {
+                        Element subprocessElement = (Element) subprocessElements.item(i);
+                        String subprocessName = subprocessElement.getElementsByTagName("name").item(0).getTextContent();
+                        String subprocessPath =
+                                subprocessElement.getElementsByTagName("path").item(0).getTextContent();
+                        String subprocessId =
+                                subprocessElement.getElementsByTagName("id").item(0).getTextContent();
+
+                        JSONObject subprocess = new JSONObject();
+                        subprocess.put("name", subprocessName);
+                        subprocess.put("path", subprocessPath);
+                        subprocess.put("id", subprocessId);
+                        subprocessArray.put(subprocess);
+                    }
+                }
+
+                if(successorElements.getLength() != 0){
+                    for(int i = 0 ; i < subprocessElements.getLength() ; i++){
```

Comment:
Fixed in https://github.com/wso2/product-pc/pull/36 (See under git comment called "Reformat the code")